week2/internal/repository/implement/film_repository.go

```diff
@@ -0,0 +1,83 @@
+package repositoryimplement
+
+import (
+	"context"
+
+	"github.com/VuKhoa23/advanced-web-be/internal/database"
+	"github.com/VuKhoa23/advanced-web-be/internal/domain/entity"
+	"github.com/VuKhoa23/advanced-web-be/internal/repository"
+	"gorm.io/gorm"
+)
+
+type FilmRepository struct {
+	db *gorm.DB
+}
+
+func NewFilmRepository(db database.Db) repository.FilmRepository {
+	return &FilmRepository{db: db}
+}
+
+func (repo *FilmRepository) GetFilmByID(ctx context.Context, id int64) (*entity.Film, error) {
+	var film entity.Film
+	result := repo.db.WithContext(ctx).First(&film, id)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	return &film, nil
+}
+
+func (repo *FilmRepository) DeleteFilm(ctx context.Context, id int64) error {
+	return repo.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
+		// Delete from film_actor
+		if err := tx.Exec(`
+            DELETE fa FROM film_actor fa
+            JOIN film f ON f.film_id = fa.film_id
+            WHERE f.film_id = ?`, id).Error; err != nil {
```

Comment:
Not sure why we need to JOIN here. Can we just delete like this
```
DELETE fa FROM film_actor fa
WHERE fa.film_id = ?`
```
And the same for below queries

Reply:
Generating a DELETE statement from MySQL resulted in queries with those JOINs baked in so I was just being extra careful by copying it 1:1

Reply:
I think this may affect to the performance and readability of these delete queries

Reply:
I'll try removing the JOINs and see what happens